src/keylessBackup/saga.test.ts

```diff
@@ -235,15 +243,61 @@ describe('keylessBackup saga', () => {
                 mockValoraKeyshareBuffer,
                 mockEncryptedMnemonic
               ),
-              mockEncryptedMnemonic,
+              mockMnemonic,
             ],
+            [call(generateKeysFromMnemonic, mockMnemonic), { privateKey: mockPrivateKey }],
+            [call(walletHasBalance, privateKeyToAddress(mockPrivateKey)), true],
+            [call(assignAccountFromPrivateKey, mockPrivateKey, mockMnemonic), mockWalletAddress],
           ])
+          .call(storeMnemonic, mockMnemonic, mockWalletAddress)
+          .call(initializeAccountSaga)
+          .put(setBackupCompleted())
+          .put(refreshAllBalances())
           .put(keylessBackupCompleted())
           .run()
         expect(ValoraAnalytics.track).toBeCalledWith('cab_handle_keyless_backup_success', {
           keylessBackupFlow: KeylessBackupFlow.Restore,
         })
       })
+      it('bails if the user does not have a balance and chooses to exit', async () => {
+        await expectSaga(handleValoraKeyshareIssued, {
+          payload: { keyshare: mockValoraKeyshare, keylessBackupFlow: KeylessBackupFlow.Restore },
+          type: valoraKeyshareIssued.type,
+        })
+          .provide([
+            [select(torusKeyshareSelector), mockTorusKeyshare],
+            [
+              call(getSecp256K1KeyPair, mockTorusKeyshareBuffer, mockValoraKeyshareBuffer),
+              {
+                privateKey: mockEncryptionPrivateKeyBuffer,
+                publicKey: mockEncryptionPublicKeyBuffer,
+              },
+            ],
+            [
+              call(getEncryptedMnemonic, {
+                encryptionPrivateKey: mockEncryptionPrivateKey,
+                encryptionAddress: mockEncryptionAddress,
+              }),
+              mockEncryptedMnemonic,
+            ],
+            [
+              call(
+                decryptPassphrase,
+                mockTorusKeyshareBuffer,
+                mockValoraKeyshareBuffer,
+                mockEncryptedMnemonic
+              ),
+              mockMnemonic,
+            ],
+            [call(generateKeysFromMnemonic, mockMnemonic), { privateKey: mockPrivateKey }],
+            [call(walletHasBalance, privateKeyToAddress(mockPrivateKey)), false],
+          ])
+          .dispatch(keylessBackupBail())
+          .run()
+        expect(ValoraAnalytics.track).toBeCalledWith('cab_handle_keyless_backup_success', {
+          keylessBackupFlow: KeylessBackupFlow.Restore,
+        })
```

Comment:
this event shouldn't be fired if the user bails right?

Reply:
I debated this. We can either throw when the user bails and then we will log a failure or we can return when the user bails and log a success. I thought it was closer to a success since we successfully restored the wallet, the user just declined to use it.

Reply:
We could alternatively return some sort of status from `handleKeylessBackupRestore` and log analytics in `handleValoraKeyshareIssued` based off the state, if we want more granular information on what the user chose. As it is now, we have no analytics indicating when a user bails, right? (we could log something when they dispatch the bail action from the dialog as well.)

Reply:
Decided to add analytics events at the component level when an action is dispatched.